src/Worker.Extensions.DurableTask/TaskOrchestrationContextExtensionMethods.cs

```diff
@@ -20,14 +24,47 @@ public static class TaskOrchestrationContextExtensionMethods
     /// <param name="context">The task orchestration context.</param>
     /// <param name="request">The DurableHttpRequest used to make the HTTP call.</param>
     /// <returns>DurableHttpResponse</returns>
-    public static Task<DurableHttpResponse> CallHttpAsync(this TaskOrchestrationContext context, DurableHttpRequest request)
+    public static async Task<DurableHttpResponse> CallHttpAsync(this TaskOrchestrationContext context, DurableHttpRequest request)
     {
         if (context is null)
         {
             throw new ArgumentNullException(nameof(context));
         }
 
-        return context.CallActivityAsync<DurableHttpResponse>(Constants.HttpTaskActivityReservedName, request);
+        DurableHttpResponse response = await context.CallActivityAsync<DurableHttpResponse>(Constants.HttpTaskActivityReservedName, request);
+        
+        while (response.StatusCode == HttpStatusCode.Accepted && request.AsynchronousPatternEnabled )
+        {
+            // If Headers is null or missing, we can't poll the Location URL, so return the response.
+            if (response.Headers is null)
+            {
+                break;
```

Comment:
Can we add a log here that will let us know that the headers were null or missing? It might help with debugging.